addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ConsolePanel.java

```diff
@@ -448,7 +452,7 @@ public void setScript(ScriptWrapper script) {
             setTabFocus();
         }
 
-        if (!isTabVisible()) {
+        if (allowFocus && !isTabVisible()) {
```

Comment:
Probably better to wrap both if statements that set focus, or just return earlier.